internal/config/config.go

```diff
@@ -302,6 +401,88 @@ func parsePeer(p peer) (*Peer, error) {
 	}, nil
 }
 
+// Parse peer autodiscovery configuration.
+//
+// BGP peer configuration can be specified in either annotations or labels. If
+// the same parameter is specified using both annotations and labels, the value
+// specified in annotations is used. If a parameter isn't specified in
+// annotations nor in labels, a default value - if configured - is used.
+//
+// If a required BGP parameter isn't specified in annotations, labels or
+// defaults, an error is returned. Required parameters:
+// - Local ASN
+// - Peer ASN
+// - Peer address
+func parsePeerAutodiscovery(p peerAutodiscovery) (*PeerAutodiscovery, error) {
+	pad := &PeerAutodiscovery{}
+
+	if p.FromAnnotations == nil && p.FromLabels == nil {
+		return nil, errors.New("missing from-annotations or from-labels")
+	}
+
+	if p.Defaults != nil {
+		pad.Defaults = &PeerAutodiscoveryDefaults{
+			ASN:     p.Defaults.ASN,
+			MyASN:   p.Defaults.MyASN,
+			Address: net.ParseIP(p.Defaults.Address),
+			Port:    p.Defaults.Port,
+		}
```

Comment:
I've more or less cloned the existing config parsing behavior which relies on an unexported struct for unmarshaling and a matching exported struct for representing the configuration in memory after parsing. I'm not sure if this is 100% necessary, however at least in some cases it can't be enough to just unmarshal, e.g. `Address` above is of type `net.IP` and AFAICT you have to call `net.ParseIP()` to parse the string provided by the user.